components/RuiModal/readme.md

```diff
@@ -0,0 +1,171 @@
+<rui-modal id="modal-two" size="medium">
+    <h2 slot="heading">Modal Heading</h2>
+    <div slot="detail">
+        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis. <a href="#">A focusable element</a></p>
+        <p><a href="#">Test</a>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis.</p>
+        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis.</p>
+        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis.</p>
+        <p><a href="#">Test test</a>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis.</p>
+    </div>
+    <rui-button slot="confirm">Confirm</rui-button>
+    <rui-button slot="cancel" variant="outline">Cancel</rui-button>
+</rui-modal>
+```
+
+## Responding to Modal actions
+The modal emits two types of events, `rui-modal-confirm` is emitted when the user click the 'confirm' slotted action, otherwise if the modal is closed for any other reason the `rui-modal-cancel` event will be emitted. 
+
+```html
+<script>
+    const el = document.getElementByID('modal-action-example')
+    if (el) {
+        el.addEventListener('rui-modal-confirm', () => {
+            alert('User confirmed')
+        })
+
+        el.addEventListener('rui-modal-cancel', () => {
+            alert('User canceled')
+        })
+    }
+</script>
+<rui-modal id="modal-action-example" size="medium">
+    <h2 slot="heading">Modal Heading</h2>
+    <div slot="detail">
+        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis.</p>
+    </div>
+    <rui-button slot="confirm">Confirm</rui-button>
+    <rui-button slot="cancel" variant="outline">Cancel</rui-button>
+</rui-modal>
+```
+
+## Changing Close Conditions
+By default the user can close a modal by esc key press or by clicking outside the modal. If you wish to disable either of these you can
+via the `no-click-outside-close` and `no-esc-btn-close` attributes
+
+<rui-button onClick={() => {
+    document.getElementById('modal-three').setAttribute('open', true);
+}}>Click outside disabled</rui-button>
+
+```html preview
+<rui-modal no-click-outside-close id="modal-three">
+    <h2 slot="heading">Modal Heading</h2>
+    <div slot="detail">
+        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis.</p>
+    </div>
+</rui-modal>
+```
+
+<rui-button onClick={() => {
```

Comment:
Can we move these buttons inside the preview so that you can see how to open the modals?